docs/reference/query-dsl/has-child-query.asciidoc

```diff
@@ -9,7 +9,7 @@ an example:
 --------------------------------------------------
 {
     "has_child" : {
-        "type" : "blog_tag",
+        "child_type" : "blog_tag",
```

Comment:
At the moment both names are valid in the parser.

type is what's documented. child_type is what's being tested for here: https://github.com/elastic/elasticsearch/blob/09d2b4130003191631857d5b8a59c22c84ea8fbb/core/src/test/java/org/elasticsearch/index/query/HasChildQueryBuilderTests.java#L200 and what's being generated here https://github.com/elastic/elasticsearch/blob/10ddd691a36fa3a8199fac8da43da540af19a697/core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java#L186 (introduced as part of this https://github.com/elastic/elasticsearch/commit/3cd54fc4ee70a395e26dc5220e8245a55e2723aa )

IMHO it would make sense to consolidate the naming and have one preferred name, what do you think? (Personally I don't care which name it is for as long as docs, json generation code, parser code and tests all agree).